yt/yt/ytlib/chunk_client/input_chunk.h

```diff
@@ -135,15 +141,33 @@ class TInputChunk
     //! Releases memory occupied by HeavyColumnarStatisticsExt.
     void ReleaseHeavyColumnarStatisticsExt();
 
+    //! Returns the approximate row count based on the row indexes specified in lower/upper limit.
     i64 GetRowCount() const;
+
+    //! Returns the approximate data weight based on the exact total chunk data weight, the row indexes specified
+    //! in lower/upper limits and the input chunks selectivity factors (see next paragraph).
+    //! Columnar factors are considered for all formats, i.e. we approximate the logical data weight bounded by all selectors.
+    //!
+    //! The data weight reduction logic is performed as follows:
+    //!   1. If `BlockSelectivityFactor` is set, we assume that it single-handedly accounts for the data weight that will be read.
+    //!   2. Otherwise, we compute the row selectivity factor as the proportion of rows bounded by limits out of all the rows in the chunk
+    //!      and use it alongisde `ColumnSelectivityFactor`.
     i64 GetDataWeight() const;
+    //! Same as above, but for uncompressed data size.
+    //! Columnar factors are only considered for columnar formats, i.e. we approximate the uncompressed data size that needs to be read from disk.
     i64 GetUncompressedDataSize() const;
+    //! Same as above, but for uncompressed data size.
+    //! Columnar factors are only considered for columnar formats, i.e. we approximate the compressed data size that needs to be read from disk.
```

Comment:
`i.e. we approximate the compressed data size that needs to be read from disk.` -> `i.e. we approximate the compressed data size that is needed to be read from disk.`